src/main/process-pool.ts

```diff
@@ -0,0 +1,70 @@
+import { ChildProcess, fork } from 'child_process';
+import * as os from 'os';
+import { resolve as pathResolve } from 'path';
+import { removeDebugArgs } from './util';
+
+interface IMochaProcess extends ChildProcess {
+  destroy: () => void;
+}
+
+export default class ProcessPool {
+  private maxParallel: number;
+  private waitingList: Array<(process: IMochaProcess) => void> = [];
+  private unusedProcesses: IMochaProcess[] = [];
+  private processes: IMochaProcess[] = [];
+
+  constructor() {
+    this.maxParallel = os.cpus().length;
+  }
+
+  setMaxParallel(n) {
```

Comment:
We're TS so let's be explicit and add both arguments types and return types. If you have time please change the tsconfig setting to fail when these values are not specified (but that's not necessary).